pyriemann/transfer/_euclidean.py

```diff
@@ -0,0 +1,190 @@
+
+    else:
+        print("Already aligned!")
+        r_ea = mean
+
+    return r_ea
+
+
+class TransformEA(BaseEstimator, TransformerMixin):
+    """Euclidean Alignment, based on [1].
+
+    The Euclidean Alignment is a transformation that recenter each domain's
+    Euclidean mean covariance matrix to the Identity, making distributions
+    more similar.
+
+    This transformation accept both raw signals and covariance matrices as
+    data inputs. For raw trials, you can determine the estimator using the
+    'estimtor' parameter.
+
+    .. note::
+       Using .fit() and then .transform() will give different results than
+       .fit_transform(). In fact, .fit_transform() should be applied on the
+       training dataset (target and source) and .transform() on the test
+       partition of the target dataset.
+
+    Parameters
+    ----------
+
+    dtype: str
+    Type of input data, it can be 'covmat' for covariance or 'raw' for raw trials
+
+    estimator: str
+    If dtype=='raw', it represents the covariance estimator
+
+    Attributes
+    ----------
+    _ref : list
+        Contains the reference matrix for each domain.
+        Order of the list is the same as 'domains'
+
+    _cov : Dict
+        Dictionary containing the covariance matrices for each domain.
+        Order of the list is the same as 'domains'
+
+    References
+    ----------
+    .. [1] `Transfer Learning for Brain-Computer Interfaces:
+        A Euclidean Space Data Alignment Approach
+        <https://arxiv.org/abs/1808.05464>`_
+        He He and Dongrui Wu, IEEE Transactions on Biomedical Engineering, 2019
+
+    """
+    def __init__(self, estimator='lwf', dtype='covmat'):
+        self.estimator = estimator
+        self.dtype = dtype
+        self._ref = list()
+        self._cov = {}
+
+    def _compute_ref(self, X):
+        ref = compute_ref_euclidean(X)
+
+        self._ref.append(ref)
+
+    def _fit_sw(self, X, domains):
+
+        for i in range(len(np.unique(domains))):
+
+            d = np.unique(domains)[i]
+
+            X_d = X[domains == d]
+
+            if self.dtype != 'covmat':
+                X_d = self._transform_cov(X_d, d)
+
+            self._compute_ref(X_d)
+
+    def _transform_sw(self, X, domains):
+        """Align each domain"""
+
+        X_align = []
+        for i in range(len(np.unique(domains))):
+
+            d = np.unique(domains)[i]
+            X_d = X[domains == d]
+            ref_d = self._ref[i]
+
+            if self.dtype == 'covmat':
+                align = ref_d @ X_d @ ref_d
+            else:
+                align = ref_d @ X_d
+            X_align.append(align)
+
+        X_align = np.concatenate(X_align)
+        return X_align
+
+    def _transform_cov(self, X, d):
+        """ Compute covariance matrices of trials in array X and save"""
+
+        cov = covariances(X, estimator=self.estimator)
+        self._cov[d] = cov
+        return cov
+
+    def fit(self, X, y_enc):
```

Comment:
`fit` method is public, it should be documented.
Something along these lines (except you accept both epochs and covariance matrices):

https://github.com/pyRiemann/pyRiemann/blob/c44b924573dc2f4b4a9efb328d18636b8baa31d5/pyriemann/tangentspace.py#L83